sphinx/ext/intersphinx/_load.py

```diff
@@ -52,22 +53,13 @@ def validate_intersphinx_mapping(app: Sphinx, config: Config) -> None:
     errors = 0
     for name, value in config.intersphinx_mapping.copy().items():
         # ensure that intersphinx projects are always named
-        if not isinstance(name, str):
+        if not isinstance(name, str) or not name:
             errors += 1
             msg = __(
                 'Invalid intersphinx project identifier `%r` in intersphinx_mapping. '
                 'Project identifiers must be non-empty strings.'
             )
-            LOGGER.error(msg % name)
-            del config.intersphinx_mapping[name]
-            continue
-        if not name:
-            errors += 1
-            msg = __(
-                'Invalid intersphinx project identifier `%r` in intersphinx_mapping. '
-                'Project identifiers must be non-empty strings.'
-            )
-            LOGGER.error(msg % name)
+            LOGGER.error(msg, name)
```

Comment:
I observed that the error was actually the same so... (and I totally missed the usage of '%').

Reply:
See #12644